samples/basic/threads/src/main.c

```diff
@@ -109,8 +109,8 @@ void uart_out(void)
 }
 
 K_THREAD_DEFINE(blink1_id, STACKSIZE, blink1, NULL, NULL, NULL,
-		PRIORITY, 0, K_NO_WAIT);
+		PRIORITY, 0, 0);
```

Comment:
Doesn't this eventually get propagated to k_thread_create()? Why does this have to be a 0 and not K_NO_WAIT?

Reply:
The timeout constructor macros are written in terms of the time unit conversion API, which are C functions.  C function calls, even pure ones that can be evaluated at compile time based on constant arguments, can't be used as a rvalue to a static initializer (C++ has constexpr for this, C never bothered).

We'd need a separate API to construct these, when it's really not that useful (the only use so far isn't to start a thread on a timer, it's to not start a thread at all).  The decision from the last time around was that we could address this with a new thread definition API (e.g. a "K_NO_START" flag).